v22.1/manage-certs-vault.md

```diff
@@ -0,0 +1,660 @@
+    else
+        echo "roach already installed"
+    fi'
+}
+
+install_roach_on_node $node1name
+install_roach_on_node $node2name
+install_roach_on_node $node3name
+```
+
+```txt
+roach not installed; installing roach
+roach not installed; installing roach
+roach not installed; installing roach
+```
+
+#### Generate and provision a start script for each node
+
+Note that each start script is node specific, configuring each node to advertise its own IP address to the cluster, and attempt to `join` a specified list of peers at specific IP addresses.
+
+{% include_cached copy-clipboard.html %}
+```shell
+# for node 1
+cat <<~script~ > start_roach.sh
+cockroach start \
+--certs-dir=certs \
+--advertise-addr="${node1addr}" \
+--join="${node1addr},${node2addr},${node3addr}" \
+--cache=.25 \
+--max-sql-memory=.25 \
+--background
+~script~
+
+chmod +x start_roach.sh
+gcloud compute scp ./start_roach.sh $node1name:~
+
+# for node 2
+cat <<~script~ > start_roach.sh
+cockroach start \
+--certs-dir=certs \
+--advertise-addr="${node2addr}" \
+--join="${node1addr},${node2addr},${node3addr}" \
+--cache=.25 \
+--max-sql-memory=.25 \
+--background
+~script~
+
+chmod +x start_roach.sh
+gcloud compute scp ./start_roach.sh $node2name:~
+
+# for node 3
+cat <<~script~ > start_roach.sh
+cockroach start \
+--certs-dir=certs \
+--advertise-addr="${node3addr}" \
+--join="${node1addr},${node2addr},${node3addr}" \
+--cache=.25 \
+--max-sql-memory=.25 \
+--background
+~script~
+
+chmod +x start_roach.sh
+gcloud compute scp ./start_roach.sh $node3name:~
+```
+
+```shell
+start_roach.sh                                                                              100%  156     3.0KB/s   00:00
+start_roach.sh                                                                              100%  156     3.0KB/s   00:00
+start_roach.sh                                                                              100%  156     3.1KB/s   00:00
+```
+
+### Start or reset the cluster
+
+#### Start the nodes
+
+To start the cluster, execute the pre-provisioned start script.
+
+{% include_cached copy-clipboard.html %}
+```shell
+gcloud beta compute ssh --command './start_roach.sh' "${node1name}"
+gcloud beta compute ssh --command './start_roach.sh' "${node2name}"
+gcloud beta compute ssh --command './start_roach.sh' "${node3name}"
+````
+
+#### Reset the nodes
+
+To cause an already running cluster to switch to the new credentials, send a `SIGHUP` signal to each node.
+
+{% include_cached copy-clipboard.html %}
+```shell
+gcloud beta compute ssh "$node1name" --command 'pkill -SIGHUP -x cockroach'
+gcloud beta compute ssh "$node2name" --command 'pkill -SIGHUP -x cockroach'
+gcloud beta compute ssh "$node3name" --command 'pkill -SIGHUP -x cockroach'
+```
+
+## Client operations
+
+### Provision the client
+
+#### Install CockroachDB
+
+Install Cockroach locally [another way](install-cockroachdb-mac.html) or with docker as follows:
```

Comment:
```suggestion
Instead of downloading the binary, you can install CockroachDB using Docker:

~~~shell
docker pull cockroachdb/cockroach:latest
~~~


You can also [install the `cockroach` binary locally on macOS](install-cockroachdb-mac.html).
```
If you apply this, change the tildes above to backticks. I couldn't get it to work with backticks inside backticks.